libs/core/src/main/java/org/opensearch/core/common/StopWatch.java

```diff
@@ -30,7 +30,7 @@
  * GitHub history for details.
  */
 
-package org.opensearch.common;
+package org.opensearch.core.common;
 
 import org.opensearch.common.unit.TimeValue;
 import org.opensearch.core.common.lease.Releasable;
```

Comment:
What do you think about the following?

- Remove the `Releasable` import
- Change the `timing()` method (line 157) to the following:

```
public TimingHandle timing(String taskName) {
    start(taskName);
    return this::stop;
}

public interface TimingHandle extends AutoCloseable {
    @Override
    void close();
}
```

- Move this class to `libs/common` instead of `libs/core`

As far as I can tell there should be no OpenSearch core-specific functionality in this utility class so `common` is a better place for it. Also, only ClusterApplierService uses the `timing()` method so my proposed refactoring will be less intrusive than moving StopWatch to core.

Reply:
Alternatively, `Releasable` could be moved into `lib/common`. The only "core" input in that class is in the Javadoc where it [incorrectly states that it can only throw a BaseOpenSearchException](https://github.com/opensearch-project/OpenSearch/blob/db0933b58d10291e5cfe6a95badbe6810d86ffed/libs/core/src/main/java/org/opensearch/core/common/lease/Releasable.java#L38) (in fact it can throw any unchecked exception).

Reply:
I actually like both options. It's nice to decouple Releasable from StopWatch, and then pull them both to the common library.

Reply:
done